### PR TITLE
feat: Company crud

### PR DESCRIPTION
**Pull request by iraboldyrieva**
## Describe your changes

- Added `AddNewCompany` and `UpdateCompany` components to `CompanyListPage`
- Updated `CompanyListPage/index.tsx` to integrate new components
- Added corresponding styles and updated `i18n` for new features
- Fix bug in Driver Avatar
- Fix bug with search
- Create toast messages

## Issue ticket code (and/or) and link

- [SCRUM-61: Company CRUD](https://codecraftersintership.atlassian.net/browse/SCRUM-61)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/ModalForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import React, { FC, useState } from 'react';
+
+import { Box, IconButton, Typography } from '@mui/material';
+
+import Button from '../Button';
+import { formContainer, formWrapper } from './styles';
```
Comment (lyudad) [inline src/components/ModalForm/index.tsx]: fix order of 2 last lines

File: src/pages/CompanyListPage/AddCompany/AddNewCompany.tsx
Comment (lyudad) [inline src/pages/CompanyListPage/AddCompany/AddNewCompany.tsx:1]: add and edit popups are the same, exept query, why its different components?

File: src/pages/CompanyListPage/AddCompany/AddNewCompany.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  const authToken = useSelector((store: RootState) => store.auth.token);
+
+  const sendData = async (formData: CreateCompanyForm) => {
+    try {
+      await axiosInstance.post(
+        '/company',
+        {
+          ...formData,
+          logo: 'http://logo-sample',
```
Comment (lyudad) [inline src/pages/CompanyListPage/AddCompany/AddNewCompany.tsx]: в олега копировала? плохо. 1.1.11

File: src/pages/CompanyListPage/AddCompany/styles.css [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+/* .title {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  font-family: '"Public Sans", sans-serif';
+
+  margin-bottom: 50px;
+} */
```
Comment (lyudad) [inline src/pages/CompanyListPage/AddCompany/styles.css]: шо це таке 1.1.10

File: src/pages/CompanyListPage/UpdateCompany/UpdateCompany.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+  const authToken = useSelector((store: RootState) => store.auth.token);
+
+  const sendData = async (formData: UpdateCompanyForm) => {
+    try {
+      await axiosInstance.patch(
+        `/company/${companyId}`,
+        {
+          ...formData,
+          logo: 'http://logo-sample',
```
Comment (lyudad) [inline src/pages/CompanyListPage/UpdateCompany/UpdateCompany.tsx]: 1.1.11

File: src/pages/CompanyListPage/AddCompany/AddNewCompany.tsx
Comment (iraboldyrieva) [reply src/pages/CompanyListPage/AddCompany/AddNewCompany.tsx:1]: will do one component

File: src/pages/CompanyListPage/AddCompany/AddNewCompany.tsx (hunk 2 above)
Comment (iraboldyrieva) [reply src/pages/CompanyListPage/AddCompany/AddNewCompany.tsx]: no its Alex's code, will fix

**Review by lyudad — APPROVED** [review]

**Pull request merged**